tests/test_config.py

```diff
@@ -1095,252 +2042,241 @@ class FakeChecker:
+            ),
+            dedent(
+                """\
+                name: test-charm-name-from-metadata-yaml
+                summary: test summary
+                description: test description
+                """
+            ),
+        ],
+    ],
+)
 def test_schema_analysis_ignore_attribute_missing(
-    create_config, check_schema_error, tmp_path, create_checker
+    tmp_path,
+    create_checker,
+    prepare_charmcraft_yaml,
+    prepare_metadata_yaml,
+    charmcraft_yaml,
+    metadata_yaml,
 ):
     """An attribute specified to ignore is missing in the system."""
     create_checker("check_ok_1", linters.CheckType.attribute)
     create_checker("check_ok_2", linters.CheckType.lint)
-    create_config(
-        """
-        type: bundle  # mandatory
-        analysis:
-            ignore:
-                attributes: [check_ok_1, check_missing]
-                linters: [check_ok_2]
-    """
-    )
-    check_schema_error(
+    prepare_charmcraft_yaml(charmcraft_yaml)
+    prepare_metadata_yaml(metadata_yaml)
+
+    with pytest.raises(CraftError) as cm:
+        load(tmp_path)
+    assert str(cm.value) == dedent(
         """\
         Bad charmcraft.yaml content:
         - Bad attribute name 'check_missing' in field 'analysis.ignore.attributes[1]'"""
     )
 
 
+@pytest.mark.parametrize(
+    "charmcraft_yaml, metadata_yaml",
+    [
+        [
+            dedent(
+                """\
+                type: bundle
+                analysis:
+                  ignore:
+                    attributes: [check_ok_1]
+                    linters: [check_ok_2, check_missing]
+                """
+            ),
+            dedent(
+                """\
+                name: test-charm-name-from-metadata-yaml
+                summary: test summary
+                description: test description
+                """
+            ),
+        ],
+    ],
+)
 def test_schema_analysis_ignore_linter_missing(
-    create_config, check_schema_error, tmp_path, create_checker
+    tmp_path,
+    create_checker,
+    prepare_charmcraft_yaml,
+    prepare_metadata_yaml,
+    charmcraft_yaml,
+    metadata_yaml,
 ):
     """A linter specified to ignore is missing in the system."""
     create_checker("check_ok_1", linters.CheckType.attribute)
     create_checker("check_ok_2", linters.CheckType.lint)
-    create_config(
-        """
-        type: bundle  # mandatory
-        analysis:
-            ignore:
-                attributes: [check_ok_1]
-                linters: [check_ok_2, check_missing]
-    """
-    )
-    check_schema_error(
+    prepare_charmcraft_yaml(charmcraft_yaml)
+    prepare_metadata_yaml(metadata_yaml)
+
+    with pytest.raises(CraftError) as cm:
+        load(tmp_path)
+    assert str(cm.value) == dedent(
         """\
         Bad charmcraft.yaml content:
         - Bad lint name 'check_missing' in field 'analysis.ignore.linters[1]'"""
     )
-
-
-def test_actions_defined_in_charmcraft_yaml(create_config, tmp_path):
```

Comment:
Sorry I missed this last night - I don't see equivalents of these 3 tests still here. Can you put some up?

Reply:
Added for now, it will be moved to another test files.